### PR TITLE
[SYCL][Graph] Update UR tag for L0 kernel binary update

### PR DESCRIPTION
**Pull request by Bensuo**
- Fix potential implicit conversion from queue to command_graph when using dynamic command groups
- Remove XFAIL for dyn_cgf* tests on L0
- Fix dyn_cgf_accessor_spv test and add new prebuilt spirv binary for it

### CODE REVIEW

File: sycl/test-e2e/Graph/Update/dyn_cgf_different_arg_nums.cpp [hunk 1]
```diff
@@ -1,12 +1,9 @@
...
 // RUN: env SYCL_UR_TRACE=2 %{run} %t.out | FileCheck %s
 // Extra run to check for leaks in Level Zero using UR_L0_LEAKS_DEBUG
-// RUN: %if level_zero %{env SYCL_PI_LEVEL_ZERO_USE_IMMEDIATE_COMMANDLISTS=0 %{l0_leak_check} %{run} %t.out 2>&1 | FileCheck %s --implicit-check-not=LEAK %}
+// RUN: %if level_zero %{env SYCL_PI_LEVEL_ZERO_USE_IMMEDIATE_COMMANDLISTS=0 env SYCL_UR_TRACE=2 %{l0_leak_check} %{run} %t.out 2>&1 | FileCheck %s --implicit-check-not=LEAK %}
 // Extra run to check for immediate-command-list in Level Zero
-// RUN: %if level_zero %{env SYCL_PI_LEVEL_ZERO_USE_IMMEDIATE_COMMANDLISTS=1 %{l0_leak_check} %{run} %t.out 2>&1 | FileCheck %s --implicit-check-not=LEAK %}
-
-// XFAIL: level_zero
-// XFAIL-TRACKER: OFNAAO-307
+// RUN: %if level_zero %{env SYCL_PI_LEVEL_ZERO_USE_IMMEDIATE_COMMANDLISTS=1 env SYCL_UR_TRACE=2 %{l0_leak_check} %{run} %t.out 2>&1 | FileCheck %s --implicit-check-not=LEAK %}
```
Comment (EwanC) [inline sycl/test-e2e/Graph/Update/dyn_cgf_different_arg_nums.cpp:6]: `SYCL_UR_TRACE=2` a debugging leftover?

**Review by EwanC — APPROVED** [review]
LGTM

**Comment by EwanC** [comment]
I didn't want to disable all the L0 update testing, so didn't make this change earlier. But now could you add ` UR_DEVICE_COMMAND_BUFFER_UPDATE_CAPABILITY_FLAG_KERNEL_HANDLE` to https://github.com/intel/llvm/blob/795ff19371223bd36147a1528f9b5209a7478337/sycl/source/detail/device_impl.cpp#L722 so that it's a required UR device capability for `aspect::ext_oneapi_graph`

File: sycl/test-e2e/Graph/Update/dyn_cgf_different_arg_nums.cpp (hunk 1 above)
Comment (Bensuo) [reply sycl/test-e2e/Graph/Update/dyn_cgf_different_arg_nums.cpp]: Ah correct, thanks for catching that! It's removed now.

**Review by cperkinsintel — APPROVED** [review]

**Review by callumfare — APPROVED** [review]

**Comment by EwanC** [comment]
@intel/llvm-gatekeepers This is good to merge now, thanks

**Pull request merged**